### PR TITLE
fix(coding-agent): show a preparing turn's prompt in the TUI queued-messages area

### PR DESCRIPTION
**Pull request by snimu**
No-Ticket: RES-1306 is an internal Linear issue (https://linear.app/primeintellect/issue/RES-1306); no GitHub issue exists.

## What

A queued turn can trigger a context compaction before it starts. During that compaction the turn sits in a "preparing" state the queued-messages area never displayed, so the prompt vanished until the turn began. The area now shows it as a `Starting: …` line, from the active-action status the session already publishes.

## Why

A queued prompt that triggered a compaction vanished from the queued-messages area the moment its turn was picked up, and it stayed hidden for the entire compaction, which can run minutes on large transcripts. The area listed only steering and follow-up messages; the preparing state was reported in a field the interactive UI never rendered.

## Behavior changes

- The queued-messages area shows the prompt as a `Starting: …` line while its pre-turn compaction runs.
- The browse/edit hint still appears only for editable queued messages.
- The prompt is not shown twice: a running turn renders through the transcript, not in the queued area.
- The status label for agent-to-agent and other injected messages now uses the same short preview as the queue lanes instead of the raw message envelope; the agents view status column changes too.
- No daemon protocol change: the session-action snapshot shape is unchanged.

## Tests and validation

One regression test in `packages/coding-agent/test/interactive-mode-status.test.ts` covers the reported case: the prompt stays visible while the turn is preparing, and the queued area renders nothing once it runs. It fails without the fix. The branch merged current main; the earlier design (an extra snapshot field plus tests in the compaction suite) was dropped in favor of rendering the existing active-action status. Validation: eight affected suites pass 344/344 in a sandbox; biome, tsgo, and the testing-policy check against main pass.

### CODE REVIEW

File: packages/coding-agent/src/modes/interactive/interactive-mode.ts [hunk 1]
```diff
@@ -7501,9 +7501,26 @@ export class InteractiveMode {
...
+		// there for a long time), so the snapshot's preparing previews keep them
+		// visible. Older daemons only publish the first one via active.label.
+		const sessionActions = this.connectionState?.sessionActions;
+		const startingTurns =
+			sessionActions?.preparing ??
+			(sessionActions?.active?.kind === "turn" &&
+			sessionActions.active.phase === "preparing" &&
+			sessionActions.active.label !== undefined
+				? [sessionActions.active.label]
+				: []);
```
[the remark this answers is not in the document]
Comment (snimu) [reply packages/coding-agent/src/modes/interactive/interactive-mode.ts:7514]: Dismissing with the concrete live path. The version-match premise is wrong: clients do NOT require app/protocol/schema equality to attach — DaemonClient gates commands by capability and minProtocol/minSchemaRevision compatibility (daemon-client.ts supportsServerCapability, daemon-protocol.ts daemonSupportsCommand); the schemaId EQUALITY checks live only in daemon-launch/daemon-ps, where they classify daemons current/stale to decide UPDATES, not to refuse attachment. The path that serves a preparing-less snapshot to a current client is a routine one: a binary update with running sessions. Update-restart deliberately preserves running WORKERS on the old binary (worker descriptors do not even record appVersion; adoption via loadWorkerDescriptors/worker_auth has no version-equality requirement), the new supervisor adopts them, and the new-build TUI attaches through it to old-build workers whose session_action_update events and attach snapshots are produced by the old code — without `preparing`. The fallback covers exactly that window, per AGENTS.md's rule that optional daemon metadata must degrade locally. Renamed nothing; the test's 'legacy' case pins this mixed-version window.

**Review by sethkarten — APPROVED** [review]
Design review passed: claim-vs-diff verified, revert-probe coverage checked, edge cases and open threads assessed. LGTM.

— Prime Agent (sethkarten's agent)
